src/sonic-config-engine/tests/test_j2files.py

```diff
@@ -59,23 +69,24 @@ def create_machine_conf(self, platform, vendor):
                 'dell': 'onie',
                 'mellanox': 'onie'
                }
-        echo_cmd = "echo '{}_platform={}' | sudo tee -a /host/machine.conf > /dev/null".format(mode[vendor], platform)
+        echo_cmd1 = ["echo", '{}_platform={}'.format(mode[vendor], platform)]
+        echo_cmd2 = ["sudo", "tee", "-a", "/host/machine.conf"]
         if not os.path.exists('/host/machine.conf'):
             file_exist = False
             if not os.path.isdir('/host'):
                 dir_exist = False
-                os.system('sudo mkdir /host')
-            os.system('sudo touch /host/machine.conf')
-            os.system(echo_cmd)
+                subprocess.call(['sudo', 'mkdir', '/host'])
+            subprocess.call(['sudo', 'touch', '/host/machine.conf'])
+            getstatusoutput_noshell_pipe(echo_cmd1, echo_cmd2)
```

Comment:
Can you explain these commands?

Reply:
Please check this PR https://github.com/sonic-net/sonic-buildimage/pull/12065